gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java

```diff
@@ -177,8 +172,8 @@ public GoogleCloudStorageReadChannel(
 
     // Initialize metadata if available.
     GoogleCloudStorageItemInfo info = getInitialMetadata();
-    if (info != null) {
-      initMetadata(info);
+    if (info != null || readOptions.isFastFailOnNotFoundEnabled()) {
```

Comment:
Why is this required? IIUC, `info` can only be `null` iff fastFail is `disabled`. If `fastFail` is `enabled` then either client will get a `FileNotFoundException` or channel with NOT NULL `info`.